Source/Diagnostics/ComputeDiagFunctors/ParticleReductionFunctor.cpp

```diff
@@ -0,0 +1,132 @@
+
+#include "ParticleReductionFunctor.H"
+
+#include "Diagnostics/ComputeDiagFunctors/ComputeDiagFunctor.H"
+#include "Particles/MultiParticleContainer.H"
+#include "Particles/WarpXParticleContainer.H"
+#include "Utils/CoarsenIO.H"
+#include "WarpX.H"
+
+#include <AMReX_Array.H>
+#include <AMReX_BLassert.H>
+#include <AMReX_IntVect.H>
+#include <AMReX_MultiFab.H>
+#include <AMReX_REAL.H>
+
+using namespace amrex::literals;
+
+ParticleReductionFunctor::ParticleReductionFunctor (const amrex::MultiFab* mf_src, const int lev,
+        amrex::IntVect crse_ratio, const std::string fn_str, const int ispec, const int ncomp)
```

Comment:
It seems like `crse_ratio` was `const` in the header file's declaration:
```suggestion
        const amrex::IntVect crse_ratio, const std::string fn_str, const int ispec, const int ncomp)
```